src/hub.js

```diff
@@ -286,15 +287,41 @@ async function handleHubChannelJoined(entryManager, hubChannel, messageDispatch,
 
     scene.setAttribute("networked-scene", {
       room: hub.hub_id,
-      serverURL: process.env.JANUS_SERVER,
+      serverURL: `wss://${hub.host}`,
       debug: !!isDebug
     });
 
     while (!scene.components["networked-scene"] || !scene.components["networked-scene"].data) await nextTick();
-
     scene.components["networked-scene"]
       .connect()
       .then(() => {
+        // When reconnecting, update the server URL if necessary
+        NAF.connection.adapter.setReconnectionListeners(
+          () => {
+            const step = async () => {
+              if (!NAF.connection.adapter.isDisconnected()) return;
+
+              const currentServerURL = NAF.connection.adapter.serverUrl;
+              const newHubHost = await hubChannel.getHost();
```

Comment:
Hmm, I think this is impossible, but we may `await getHost` forever since `getHost` never rejects its promise.

Reply:
ah good point, i'll just thread the reject through on error at least